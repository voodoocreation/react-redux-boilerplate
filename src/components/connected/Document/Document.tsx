// istanbul ignore file
import { IncomingMessage } from "http";
import Document, {
  DocumentContext,
  DocumentProps,
  Head,
  Html,
  Main,
  NextScript
} from "next/document";
import * as React from "react";

interface IProps extends DocumentProps {
  locale: string;
}

export default class<P extends IProps> extends Document<P> {
  public static async getInitialProps(context: DocumentContext) {
    const initialProps = await Document.getInitialProps(context);
    const props = await context.renderPage();
    const req = context.req as IncomingMessage & {
      locale: string;
    };

    return {
      ...initialProps,
      ...props,
      locale: req.locale || "en-NZ"
    };
  }

  public render() {
    const { locale } = this.props as P;

    return (
      <Html lang={locale}>
        <Head>
          <meta charSet="UTF-8" />
          <meta
            content="width=device-width, initial-scale=1.0, maximum-scale=1.0"
            name="viewport"
          />

          <script
            dangerouslySetInnerHTML={{
              __html: `document.documentElement.classList.add("isClientRendered");`
            }}
          />
          <link
            href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
            rel="stylesheet"
          />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}
