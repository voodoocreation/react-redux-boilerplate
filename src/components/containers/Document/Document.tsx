import { IncomingMessage } from "http";
import Document, {
  Head,
  Main,
  NextDocumentContext as Context,
  NextScript
} from "next/document";
import * as React from "react";

// tslint:disable-next-line
const css = require("../../../scss/index.scss");

const Meta: React.SFC<{}> = () =>
  (
    <React.Fragment>
      <meta charSet="UTF-8" />
      <meta
        content="width=device-width, initial-scale=1.0, maximum-scale=5.0"
        name="viewport"
      />
    </React.Fragment>
  ) as React.ReactElement<any>;

export default class extends Document {
  public static async getInitialProps(context: Context) {
    const props = await context.renderPage();
    const req = context.req as IncomingMessage & {
      intlMessages: {};
      locale: string;
    };

    return {
      ...props,
      intlMessages: req.intlMessages,
      locale: req.locale
    };
  }

  public render() {
    const styles =
      process.env.NODE_ENV !== "production" ? (
        <style dangerouslySetInnerHTML={{ __html: css }} />
      ) : (
        <link
          href={`/assets/main.css?${this.props.__NEXT_DATA__.buildId}`}
          rel="stylesheet"
        />
      );

    return (
      <html lang={this.props.locale}>
        <Head>
          <script
            dangerouslySetInnerHTML={{
              __html: `document.documentElement.classList.add("isClientRendered");`
            }}
          />
          <link
            href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
            rel="stylesheet"
          />

          <Meta />

          {styles}
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </html>
    );
  }
}
