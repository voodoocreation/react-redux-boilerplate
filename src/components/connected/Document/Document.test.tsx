import { render } from "enzyme";
import merge from "lodash.merge";
import * as React from "react";

import Document from "./Document";

const setup = async (fn: any, fromTestProps?: any) => {
  const documentProps = merge(
    {
      __NEXT_DATA__: {
        buildId: "buildId",
        page: "/",
        pathname: "pathname"
      },
      assetPrefix: "/assetPrefix",
      dynamicImports: [],
      files: [],
      isServer: true,
      renderPage: async () => ({}),
      req: {
        intlMessages: {},
        locale: "en-NZ"
      }
    },
    fromTestProps
  );
  const initialProps = await Document.getInitialProps(documentProps);
  const props = {
    ...documentProps,
    ...initialProps
  };

  return {
    props,
    wrapper: fn(<Document {...props} />)
  };
};

const g: any = global;

describe("[connected] <Document />", () => {
  beforeEach(() => {
    process.env.NODE_ENV = "test";
    g.isServer = true;
  });

  it("renders correctly when NODE_ENV=development", async () => {
    process.env.NODE_ENV = "development";
    const { wrapper } = await setup(render);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders currectly when NODE_ENV=production", async () => {
    process.env.NODE_ENV = "production";
    const { wrapper } = await setup(render);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders currectly when locale is missing", async () => {
    const { wrapper } = await setup(render, {
      req: { locale: "" }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
