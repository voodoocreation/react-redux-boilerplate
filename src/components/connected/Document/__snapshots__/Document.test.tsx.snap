// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[containers] <Document /> renders correctly when NODE_ENV=development 1`] = `
Array [
  <link
    as="script"
    href="undefined/_next/static/buildId/pagespathname.js"
    rel="preload"
  />,
  <link
    as="script"
    href="undefined/_next/static/buildId/pages/_app.js"
    rel="preload"
  />,
  <link
    as="script"
    href="undefined/_next/static/buildId/pages/_error.js"
    rel="preload"
  />,
  null,
  <link
    href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
    rel="stylesheet"
  />,
  <meta
    charset="UTF-8"
  />,
  <meta
    content="width=device-width, initial-scale=1.0, maximum-scale=5.0"
    name="viewport"
  />,
  null,
  <div
    id="__next"
  />,
  null,
  null,
  null,
  null,
]
`;

exports[`[containers] <Document /> renders currectly when NODE_ENV=production 1`] = `
Array [
  <link
    as="script"
    href="undefined/_next/static/buildId/pagespathname.js"
    rel="preload"
  />,
  <link
    as="script"
    href="undefined/_next/static/buildId/pages/_app.js"
    rel="preload"
  />,
  <link
    as="script"
    href="undefined/_next/static/buildId/pages/_error.js"
    rel="preload"
  />,
  null,
  <link
    href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
    rel="stylesheet"
  />,
  <meta
    charset="UTF-8"
  />,
  <meta
    content="width=device-width, initial-scale=1.0, maximum-scale=5.0"
    name="viewport"
  />,
  <link
    href="/assets/main.css?buildId"
    rel="stylesheet"
  />,
  <div
    id="__next"
  />,
  null,
  null,
  null,
  null,
]
`;
